opensrp-app/src/test/java/org/smartregister/service/HTTPAgentTest.java

```diff
@@ -560,4 +566,342 @@ public void testOauth2authenticateRefreshTokenInvokesOauth2authenticateCoreWithC
+        Assert.assertNotNull(url);
+
+        HTTPAgent httpAgentSpy = Mockito.spy(httpAgent);
+
+        Mockito.doReturn(httpsURLConnection).when(httpAgentSpy).getHttpURLConnection(USER_DETAILS_ENDPOINT);
+
+        Mockito.doReturn(errorStream).when(httpsURLConnection).getInputStream();
+        Mockito.doReturn(HttpURLConnection.HTTP_INTERNAL_ERROR).when(httpsURLConnection).getResponseCode();
+
+        PowerMockito.mockStatic(IOUtils.class);
+        PowerMockito.when(IOUtils.toString(errorStream)).thenReturn("<html><p><b>message</b> Oops, something went wrong </u></p></html>");
+
+        LoginResponse loginResponse = httpAgentSpy.fetchUserDetails(USER_DETAILS_ENDPOINT, SAMPLE_TEST_TOKEN);
+
+        Assert.assertNotNull(loginResponse);
+        Assert.assertNotNull(loginResponse.message());
+        Assert.assertEquals("Dristhi login failed. Try later", loginResponse.message());
+        Assert.assertNull(loginResponse.payload());
+
+        Assert.assertEquals("UNKNOWN_RESPONSE", loginResponse.name());
+
+    }
+
+
+    @Test
+    public void testFetchUserDetailsConstructsCorrectResponseForMalformedURLRequests() throws Exception {
+
+        URL url = PowerMockito.mock(URL.class);
+        Assert.assertNotNull(url);
+
+        HTTPAgent httpAgentSpy = Mockito.spy(httpAgent);
+
+        Mockito.doReturn(httpsURLConnection).when(httpAgentSpy).getHttpURLConnection(USER_DETAILS_ENDPOINT);
+        Mockito.doThrow(new MalformedURLException()).when(httpsURLConnection).getResponseCode();
+
+        LoginResponse loginResponse = httpAgentSpy.fetchUserDetails(USER_DETAILS_ENDPOINT, SAMPLE_TEST_TOKEN);
+
+        Assert.assertNotNull(loginResponse);
+        Assert.assertNotNull(loginResponse.message());
+        Assert.assertEquals("Incorrect url", loginResponse.message());
+        Assert.assertNull(loginResponse.payload());
+
+        Assert.assertEquals("MALFORMED_URL", loginResponse.name());
+
+    }
+
+    @Test
+    public void testFetchUserDetailsConstructsCorrectResponseForConnectionTimedOutRequests() throws Exception {
+
+        URL url = PowerMockito.mock(URL.class);
+        Assert.assertNotNull(url);
+
+        HTTPAgent httpAgentSpy = Mockito.spy(httpAgent);
+
+        Mockito.doReturn(httpsURLConnection).when(httpAgentSpy).getHttpURLConnection(USER_DETAILS_ENDPOINT);
+        Mockito.doThrow(new SocketTimeoutException()).when(httpsURLConnection).getResponseCode();
+
+        LoginResponse loginResponse = httpAgentSpy.fetchUserDetails(USER_DETAILS_ENDPOINT, SAMPLE_TEST_TOKEN);
+
+        Assert.assertNotNull(loginResponse);
+        Assert.assertNotNull(loginResponse.message());
+        Assert.assertEquals("The server could not be reached. Try again", loginResponse.message());
+        Assert.assertNull(loginResponse.payload());
+
+        Assert.assertEquals("TIMEOUT", loginResponse.name());
+
+    }
+
+
+    @Test
+    public void testFetchUserDetailsConstructsCorrectResponseForRequestsWithNetworkConnectivity() throws Exception {
```

Comment:
Spelling to `WithoutNetworkConnectivity`

Reply:
Fixed